dataset.py

```diff
@@ -108,6 +108,33 @@ def random_subseq(sequence, seq_len):
     index = random.randint(0, len(sequence) - 1 - seq_len)
     return sequence[index:index + seq_len]
 
+def stretch_sequence(sequence, stretch_multiplier):
+    """ Iterate through sequence and stretch each time shift event by a factor """
+    time_sum = 0
+    time_count = 0
+    i = 0
+    while i < len(sequence):
+        evt = sequence[i]
+        if evt >= TIME_OFFSET and evt < VEL_OFFSET:
+            # Convert time event to number of seconds
+            time_sum += convert_time_evt_to_sec(evt)
+            time_count += 1
+            # TODO: Edge case where last event is time shift event
+        else:
+            if i > 0:
+                # Once there is a non time shift event, go backwards and recalculate previous sequential time shift events
+                for j in range(i - 1, i - 1 -time_count, -1):
+                    stretch_time = (convert_time_evt_to_sec(sequence[j]) / time_sum) * (stretch_multiplier * time_sum)
+                    stretch_ticks = round(stretch_time * TICKS_PER_SEC)
```

Comment:
This method of stretching is not as accurate as it could be. See `midi_io` in midi encode/decode where I wrote a quantization algorithm to convert time in seconds into events. Note that the number of time events after the stretch may increase OR decrease. Example:

We have 3 possible bins, 10 ms, 20 ms and 30 ms.
You have sequence [wait 30, wait 20]
Now you stretch by 2x. So you effectively want wait 50.
This leads to the sequence [wait 30, wait 30, wait 30, wait 10]

The ordering is always the largest wait time first.